crates/ty_python_semantic/src/module_resolver/module.rs

```diff
@@ -70,17 +94,26 @@ impl std::fmt::Debug for Module {
             .field("kind", &self.kind())
             .field("file", &self.file())
             .field("search_path", &self.search_path())
+            .field("known", &self.known())
             .finish()
     }
 }
 
 #[derive(PartialEq, Eq, Hash)]
-struct ModuleInner {
-    name: ModuleName,
-    kind: ModuleKind,
-    search_path: SearchPath,
-    file: File,
-    known: Option<KnownModule>,
+enum ModuleInner {
+    /// A module that resolves to a file (`lib.py` or `package/__init__.py`)
+    FileModule {
+        name: ModuleName,
+        kind: ModuleKind,
+        search_path: SearchPath,
+        file: File,
+        known: Option<KnownModule>,
```

Comment:
it's possible we might want namespace packages without `File`s to be recognized as `KnownModule`s in the future? But I can't think of a use case right now, so we can probably figure that out if and when it actually comes up.

Reply:
That's possible if we want to special case some library. I haven't looked at `KnonModule` itself but I suspect that it won't be too hard to add but I don't see a need to do it today (and agree with you that we can do it in the future)